### ISSUE TITLE
TCH rules are stricter than flake8-type-checking, but not necessarily better

### ISSUE THREAD

**Issue by GabDug** [charliermarsh/ruff#2193]
TCH rules currently produce much more warnings than flake8-type-checking, but may be less relevant.

Consider this snippet:
```python
from pkg import A,B

def test(value: A):
   return B()
```
With ruff, you'll get a warning. However, the flake8 plugin will not trigger: adding one of the two imports in the TYPE_CHECKING block would not prevent the module from being imported at runtime, at the extra cost of 3 more lines and splitted imports.

The same applies to TCH003: import from `typing` don't benefit much, as we already import it for `TYPE_CHECKING`
```python
from typing import TYPE_CHECKING, Any

if TYPE_CHECKING:
   ...

a: Any = "str"
```

I feel like the having flake8 way would be beneficial, either by default or with an option flag.

Anyway, thanks to all the ruff contributors, this is an incredible tool!

Tested on ruff 0.0.235

**Comment by GabDug** [comment]
I'm realising my comment may not be very clear:

TLDR: TCH warnings for types imported outside of TYPE_CHECKING should probably be ignored (by default or opt-in) if there is another import from the same module that is required out of type checking, as it's more verbose with no changes to imported modules at runtime.

**Comment by charliermarsh** [comment]
I agree with this! Would be a good change. It may not've come up with the existing flake8-type-checking test suite (which I ported over :))

**Comment by charliermarsh** [comment]
Gonna call this a bug since it's a clear deviation from "upstream".

**Comment by GabDug** [comment]
@charliermarsh it's indeed a bug, as it is different from the plugin default configuration.

However, it looks like there are options to change these behaviours in the original plugin:
https://github.com/snok/flake8-type-checking#strict

**Comment by charliermarsh** [comment]
Part of #2195, but I'll leave this open too. I'll do `strict` first.
